include/openmc/constants.h

```diff
@@ -113,296 +113,257 @@ constexpr int NUCLIDE_NONE  {-1};
 
 // Maximum number of partial fission reactions
 constexpr int PARTIAL_FISSION_MAX {4};
 
 // Resonance elastic scattering methods
-// TODO: Convert to enum
 enum class ResScatMethod {
   rvs, // Relative velocity sampling
   dbrc, // Doppler broadening rejection correction
   cxs // Constant cross section
 };
 
-// Electron treatments
-// TODO: Convert to enum
-constexpr int ELECTRON_LED {1}; // Local Energy Deposition
-constexpr int ELECTRON_TTB {2}; // Thick Target Bremsstrahlung
+enum class ElectronTreatment {
+  LED, // Local Energy Deposition
+  TTB  // Thick Target Bremsstrahlung
+};
 
 // ============================================================================
 // MULTIGROUP RELATED
 
-// MGXS Table Types
-// TODO: Convert to enum
-constexpr int MGXS_ISOTROPIC {1}; // Isotroically weighted data
-constexpr int MGXS_ANGLE {2};     // Data by angular bins
-
 // Flag to denote this was a macroscopic data object
 constexpr double MACROSCOPIC_AWR {-2.};
 
 // Number of mu bins to use when converting Legendres to tabular type
 constexpr int DEFAULT_NMU {33};
 
 // Mgxs::get_xs enumerated types
-// TODO: Convert to enum
-constexpr int MG_GET_XS_TOTAL              {0};
-constexpr int MG_GET_XS_ABSORPTION         {1};
-constexpr int MG_GET_XS_INVERSE_VELOCITY   {2};
-constexpr int MG_GET_XS_DECAY_RATE         {3};
-constexpr int MG_GET_XS_SCATTER            {4};
-constexpr int MG_GET_XS_SCATTER_MULT       {5};
-constexpr int MG_GET_XS_SCATTER_FMU_MULT   {6};
-constexpr int MG_GET_XS_SCATTER_FMU        {7};
-constexpr int MG_GET_XS_FISSION            {8};
-constexpr int MG_GET_XS_KAPPA_FISSION      {9};
-constexpr int MG_GET_XS_PROMPT_NU_FISSION  {10};
-constexpr int MG_GET_XS_DELAYED_NU_FISSION {11};
-constexpr int MG_GET_XS_NU_FISSION         {12};
-constexpr int MG_GET_XS_CHI_PROMPT         {13};
-constexpr int MG_GET_XS_CHI_DELAYED        {14};
+enum class MgxsType {
+  TOTAL,
+  ABSORPTION,
+  INVERSE_VELOCITY,
+  DECAY_RATE,
+  SCATTER,
+  SCATTER_MULT,
+  SCATTER_FMU_MULT,
+  SCATTER_FMU,
+  FISSION,
+  KAPPA_FISSION,
+  PROMPT_NU_FISSION,
+  DELAYED_NU_FISSION,
+  NU_FISSION,
+  CHI_PROMPT,
+  CHI_DELAYED
+};
 
 // ============================================================================
 // TALLY-RELATED CONSTANTS
 
-// Tally result entries
-constexpr int RESULT_VALUE  {0};
-constexpr int RESULT_SUM    {1};
-constexpr int RESULT_SUM_SQ {2};
-
-// Tally type
-// TODO: Convert to enum
-constexpr int TALLY_VOLUME          {1};
-constexpr int TALLY_MESH_SURFACE    {2};
-constexpr int TALLY_SURFACE         {3};
-
-// Tally estimator types
-// TODO: Convert to enum
-constexpr int ESTIMATOR_ANALOG      {1};
-constexpr int ESTIMATOR_TRACKLENGTH {2};
-constexpr int ESTIMATOR_COLLISION   {3};
-
-// Event types for tallies
-// TODO: Convert to enum
-constexpr int EVENT_SURFACE {-2};
-constexpr int EVENT_LATTICE {-1};
-constexpr int EVENT_KILL    {0};
-constexpr int EVENT_SCATTER {1};
-constexpr int EVENT_ABSORB  {2};
+enum class TallyResult {
```

Comment:
I wouldn't mind keeping this one as a plain enum (and perhaps retaining `RESULT_` in the member names to avoid any potential name conflicts) since this values are primarily used as indices into arrays.

Reply:
Makes sense to me, but the same meaning is retained either way. The only annoying thing is that xtensor doesn't correctly realize that this is something which should be cast to int.

Anyways, I'm just going to leave it as-is since it's not like these core tallying routines are going to be changing terribly much any time soon. Now, if there were something where static_casts had to be written any time a new tally was developed, I would indeed take the enum class status away. But for now, I am advocating to keep it just to keep everything stylistically consistent, even if it does require a few static_casts.